static/app/components/events/interfaces/crashContent/exception/useSourceMapDebug.tsx

```diff
@@ -0,0 +1,108 @@
+import uniqBy from 'lodash/uniqBy';
+
+import type {ExceptionValue, Frame} from 'sentry/types';
+import {defined} from 'sentry/utils';
+import {QueryKey, useQuery, UseQueryOptions} from 'sentry/utils/queryClient';
+
+export interface SourceMapDebugResponse {
+  errors: SourceMapProcessingIssueResponse[];
+}
+
+interface SourceMapProcessingIssueResponse {
+  data: Record<string, string>;
+  message: string;
+  type: SourceMapProcessingIssueType;
+}
+
+export enum SourceMapProcessingIssueType {
+  UNKNOWN_ERROR = 'unknown_error',
+  MISSING_RELEASE = 'no_release_on_event',
+  MISSING_USER_AGENT = 'no_user_agent_on_release',
+  MISSING_SOURCEMAPS = 'no_sourcemaps_on_release',
+  URL_NOT_VALID = 'url_not_valid',
+}
+
+const sourceMapDebugQuery = ({
+  orgSlug,
+  projectSlug,
+  eventId,
+  frameIdx,
+  exceptionIdx,
+}: UseSourceMapDebugProps): QueryKey => [
+  `/projects/${orgSlug}/${projectSlug}/events/${eventId}/source-map-debug/`,
+  {
+    query: {
+      frame_idx: `${frameIdx}`,
+      exception_idx: `${exceptionIdx}`,
+    },
+  },
+];
+
+interface UseSourceMapDebugProps {
+  eventId?: string;
+  exceptionIdx?: number;
+  frameIdx?: number;
+  orgSlug?: string;
+  projectSlug?: string | undefined;
```

Comment:
@scttcper why do we make all of these optional? Also nit, you don't need the `| undefined` for `projectSlug`.